library/core/src/panicking.rs

```diff
@@ -85,9 +84,9 @@ fn panic_bounds_check(index: usize, len: usize) -> ! {
     panic!("index out of bounds: the len is {} but the index is {}", len, index)
 }
 
-#[cfg(not(bootstrap))]
 #[cold]
 #[inline(never)]
+#[cfg_attr(bootstrap, track_caller)]
```

Comment:
Why is this necessary? Why only for the bootstrap compiler?

Reply:
See the CI failures - something in the bootstrap compiler is expecting to pass caller information to this function, but that has been dropped in non-bootstrap compilers.

I haven't deeply investigated but I suspect it's related to recent work here

Reply:
Why should non-bootstrap compilers not use it? Not using `#[track_caller]` makes the `Location::caller()` below point to this function, which is not very useful.

Reply:
Sounds like that feature is somewhat broken then, and wasn't properly tested. In fact, it looks like it never had `#[track_caller]` before? Personally I would probably open an issue (pinging everyone involved) and leave a FIXME comment here - while this is bad, it doesn't really make sense to block a bootstrap bump on it.

Reply:
I filed https://github.com/rust-lang/rust/issues/94337 and have pushed a commit adding a FIXME.